### PR TITLE
Add avifIsAlpha() to internal.h

### PR DESCRIPTION
**Pull request by y-guyon**

### CODE REVIEW

File: src/write.c [hunk 1]
```diff
@@ -2272,15 +2280,16 @@ static avifResult avifRWStreamWriteProperties(avifItemPropertyDedup * const dedu
             hasPixi = AVIF_FALSE;
         }
 #endif
+        const avifBool isAlpha = avifIsAlpha(item->itemCategory);
+        uint8_t depth = (uint8_t)itemMetadata->depth;
```
Comment (vrabaud) [inline src/write.c:2284]: const

File: src/write.c [hunk 2]
```diff
@@ -1662,18 +1667,20 @@ static avifResult avifEncoderAddImageInternal(avifEncoder * encoder,
                 }
                 cellImage = paddedCellImage;
             }
-            const int quantizer = (item->itemCategory == AVIF_ITEM_ALPHA) ? encoder->data->quantizerAlpha
+
+            const avifBool isAlpha = avifIsAlpha(item->itemCategory);
+            int quantizer = isAlpha ? encoder->data->quantizerAlpha
```
Comment (vrabaud) [inline src/write.c:1672]: const

File: src/write.c [hunk 3]
```diff
@@ -1192,7 +1192,12 @@ static avifBool avifEncoderDataShouldForceKeyframeForAlpha(const avifEncoderData
 
 static avifResult avifGetErrorForItemCategory(avifItemCategory itemCategory)
 {
-    return (itemCategory == AVIF_ITEM_ALPHA) ? AVIF_RESULT_ENCODE_ALPHA_FAILED : AVIF_RESULT_ENCODE_COLOR_FAILED;
+#if defined(AVIF_ENABLE_EXPERIMENTAL_GAIN_MAP)
+    if (itemCategory == AVIF_ITEM_GAIN_MAP) {
```
Comment (vrabaud) [inline src/write.c:1196]: that's a bug fix then?

**Review by vrabaud — APPROVED** [review]

File: src/write.c (hunk 1 above)
Comment (y-guyon) [reply src/write.c]: It will be removed in the final PR but done.

File: src/write.c (hunk 2 above)
Comment (y-guyon) [reply src/write.c]: It will be removed in the final PR but done.

File: src/write.c (hunk 3 above)
Comment (y-guyon) [reply src/write.c:1196]: Yes, a tiny one

Comment (wantehchang) [reply src/write.c:1196]: Please mention this bug fix in the commit message.

**Review by wantehchang — APPROVED** [review]
LGTM. The title of this pull request "Move avifIsAlpha() to internal.h" implies avifIsAlpha  already exists. But it is a new function added in this PR. How about changing the title to "Add avifIsAlpha() to internal.h"?

**Pull request merged**